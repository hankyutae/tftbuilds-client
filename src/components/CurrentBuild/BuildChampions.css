.build-display-champions{
  background-color:antiquewhite;
  position:relative;
  padding:1px 10px;
  padding-bottom:0.5em;
}

.build-display-champion-cards-box{
  position:relative;/* 
  display:flex;
  flex-wrap:wrap; */
  min-height:180px;
  margin: 15px 0px;
  background-color:antiquewhite;
  color: #5A1717;
  font-weight:500;
  overflow-x:auto;
}

.current-build-page-save-build{
  color:darkblue;
  font-weight:550;
  border:0;
  padding:1rem;
  border-radius:5px;
  width:100%;
  margin:auto;
  margin-bottom:1rem;
  background-color:#A2B9B1;
  display:block;

}

.current-build-page-new-build{
  position:absolute;
  top:10px;
  right:10px;
  color:red;
  font-weight:550;
  border:0;
  padding:0.5rem;
  border-radius:5px;
  margin-bottom:1rem;
  background-color:#A2B9B1;
}


.building-notice{
  max-width:400px;
  padding:1rem;
  border-radius:5px;
  margin:auto;
  background-color:#7A7861;
  color:rgb(19, 49, 88);
  font-weight:550;
}

@media screen and (min-width:550px){
  .current-build-page-save-build{
    width:400px;
  }
}
/* .headertwo-undertext{
  font-size:80%;
  display:flex;
  font-weight:500;
  justify-content:space-evenly;
  align-items: center;
} */
