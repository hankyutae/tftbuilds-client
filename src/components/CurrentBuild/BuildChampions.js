import React from 'react';
import './BuildChampions.css'
import ChampCard from './ChampCard';
//import TokenService from '../../services/token-service'
//import {Link} from 'react-router-dom';
import TokenService from '../../services/token-service';
import { Link } from 'react-router-dom';

class BuildChampions extends React.Component {
  constructor() {
    super();
    this.state = { screenSize: 0 };
    this.resize = this.resize.bind(this);
  }
  componentDidMount() {
    window.addEventListener("resize", this.resize);
    this.resize();
  }
  componentWillUnmount() {
    window.removeEventListener("resize", this.resize)
  }

  resize() {
    let currentscreenSize = (window.innerWidth >= 760) ? (window.innerWidth >= 1400) ? 2 : 1 : 0;
    if (currentscreenSize !== this.state.screenSize) {
      this.setState({ screenSize: currentscreenSize });
    }
  }


  render() {
    return (
      <div className="build-display-champions">
        <h3 className='champion-header'>
          Champions
          {/* 
          <div className='headertwo-undertext'> */}
          {/* <button className='current-build-page-new-build' onClick={this.props.newBuild}>
              New Build
            </button>
            {TokenService.hasAuthToken() ?
              <button className='current-build-page-save-build' onClick={this.props.saveBuild}>Save Build</button>
              :
              <button className='current-build-page-save-build'>
                <Link className='nav-link' to='/login'>
                  Log in To Save!
                </Link>
              </button>
            } */}

          { /* 
            </div> */}

        </h3>
        <button className='current-build-page-new-build button-cursor' onClick={this.props.newBuild}>
          Clear
        </button>
        {(!this.props.champs || this.props.champs.length===0) &&
          <div className='building-notice'>
            Begin by adding champions from below <br/> "<span style={{color:'#ffffff'}}>Add a Champion</span>" <br/> by clicking on the champ icon, or the card.<br></br>
            <span style={{color:'#ffffff'}}>Note!</span> There are 2 views to choose from when adding champions.
          </div>


        }
        <div className='build-display-champion-cards-box'>
          {
            this.props.champs &&
            this.props.champs.map((champInfo, index) =>
            <ChampCard key={index} index={index} champ={this.props.allChamps[champInfo.id]} syn={this.props.syn} screenSize={this.state.screenSize} />
          )}


        </div>
        {TokenService.hasAuthToken() ?
          <button className='current-build-page-save-build button-cursor' onClick={this.props.saveBuild}>Save Build</button>
          :
            <Link className='nav-link current-build-page-save-build button-cursor' to='/login'>
              Log in To Save!
            </Link>
        }
      </div >
    );
  }

}

export default BuildChampions;