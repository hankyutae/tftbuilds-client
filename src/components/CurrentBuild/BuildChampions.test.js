import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import BuildChampions from './BuildChampions';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <BuildChampions/>
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create( 
      <BrowserRouter>
        <BuildChampions/>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});