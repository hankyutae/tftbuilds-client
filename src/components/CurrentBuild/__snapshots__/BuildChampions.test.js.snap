// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="build-display-champions"
>
  <h3
    className="champion-header"
  >
    Champions
  </h3>
  <button
    className="current-build-page-new-build button-cursor"
  >
    Clear
  </button>
  <div
    className="building-notice"
  >
    Begin by adding champions from below 
    <br />
     "
    <span
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      Add a Champion
    </span>
    " 
    <br />
     by clicking on the champ icon, or the card.
    <br />
    <span
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      Note!
    </span>
     There are 2 views to choose from when adding champions.
  </div>
  <div
    className="build-display-champion-cards-box"
  />
  <a
    className="nav-link current-build-page-save-build button-cursor"
    href="/login"
    onClick={[Function]}
  >
    Log in To Save!
  </a>
</div>
`;
